drivers/hmis/app/models/hmis/form/definition.rb

```diff
@@ -40,6 +40,8 @@ class Hmis::Form::Definition < ::GrdaWarehouseBase
   # convenience attr for passing graphql args
   attr_accessor :filter_context
 
+  validates :identifier, format: { with: /\A[a-zA-Z][a-zA-Z0-9_-]*\z/, message: 'must contain only alphanumeric characters, underscores, and dashes, and must start with a letter' }
```

Comment:
This looks good but maybe we can restrict this further to make our lives easier:
* do we need to support dashes? Dashes are invalid in an identifier (variable name)
* perhaps we don't need mixed-case? Keeping everything lowercase is easier to handle

We can work around these of course with transformations or quoting. If we already have definitions with loosey-goosey identifiers it's probably not worth it to fix. But if not, I think it would be better if the identifiers were 1:1 with the reporting table names.

Reply:
It looks like we do already have many (mostly?) loosey goosey identifiers, unfortunately.

(in client prod,
```
Hmis::Form::Definition
  .where("identifier ~* ?", '([A-Z]|-)') # capitals or dashes
  .pluck(:identifier)
  .uniq.count
=> 71
Hmis::Form::Definition.pluck(:identifier).uniq.count
=> 71
```
)